### PR TITLE
[consolidate events] execution steps

### PR DESCRIPTION
**Pull request by alangenfeld**
Create a new "dagster event" in the logging based system and consolidate the currently yielded execution step events in to that, removing the top level logging based ones.

### CODE REVIEW

File: python_modules/dagster/dagster/core/events/execution.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    ):
+        return super(ExecutionStepEvent, cls).__new__(
+            cls,
+            check.str_param(event_type_value, 'event_type_value'),
+            check.str_param(step_key, 'step_key'),
+            check.str_param(pipeline_name, 'pipeline_name'),
+            check.str_param(solid_name, 'solid_name'),
+            check.str_param(solid_definition_name, 'solid_definition_name'),
+            check.str_param(step_kind_value, 'step_kind_value'),
+            check.opt_inst_param(step_output_data, 'step_output_data', StepOutputData),
```
Comment (schrockn) [inline python_modules/dagster/dagster/core/events/execution.py]: no need to do this now, but i think we will want to change step_output_data, step_failure_data, and step_success_data to be one `event_specific_data` property that is a union.

File: python_modules/dagster/dagster/utils/error.py [hunk 2]
```diff
@@ -2,11 +2,13 @@
 
 import traceback
 
-SerializableErrorInfo = namedtuple('SerializableErrorInfo', 'message stack')
+SerializableErrorInfo = namedtuple('SerializableErrorInfo', 'message stack cls_name')
```
Comment (schrockn) [inline python_modules/dagster/dagster/utils/error.py]: feel free to make this a "typed" named tuple (overriding __new__) if you are feeling frisky

File: python_modules/dagster/dagster/core/events/__init__.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    @property
+    def step_output_data(self):
+        assert (
+            self.event_type == DagsterEventType.STEP_OUTPUT
+        ), 'step_output_data only available on STEP_OUTPUT'
+        return self.event_specific_data
+
+    @property
+    def step_success_data(self):
+        assert (
```
Comment (schrockn) [inline python_modules/dagster/dagster/core/events/__init__.py]: let's use check here

Comment (alangenfeld) [reply python_modules/dagster/dagster/core/events/__init__.py]: did this further up the stack

**Pull request merged**